code/modules/mob/living/simple_animal/friendly/dog.dm

```diff
@@ -523,17 +523,18 @@
 /mob/living/simple_animal/pet/dog/corgi/narsie/Life()
 	..()
 	for(var/mob/living/simple_animal/pet/P in range(1, src))
-		if(P != src && !istype(P,/mob/living/simple_animal/pet/dog/corgi/narsie))
-			visible_message("<span class='warning'>[src] devours [P]!</span>", \
-			"<span class='cult big bold'>DELICIOUS SOULS</span>")
-			playsound(src, 'sound/magic/demon_attack1.ogg', 75, TRUE)
-			narsie_act()
-			if(P.mind)
-				if(P.mind.hasSoul)
-					P.mind.hasSoul = FALSE //Nars-Ian ate your soul; you don't have one anymore
-				else
-					visible_message("<span class='cult big bold'>... Aw, someone beat me to this one.</span>")
-			P.gib()
+		if(!health <= 0)
```

Comment:
move this check to before iterating through mobs in range, and fix it -- i'm asssuming you meant ``if (health > 0)``

```dm
..()
if (health <= 0)
	return
for (var/mob/living/simple_animal/pet/P in range(1, src))
	// stuff here
```